src/exo/API_scheduling.py

```diff
@@ -1884,6 +1884,34 @@ def merge_writes(proc, block_cursor):
     return Procedure(ir, _provenance_eq_Procedure=proc, _forward=fwd)
 
 
+@sched_op([AssignOrReduceCursorA])
+def split_write(proc, stmt):
+    """
+    Split a reduce or assign statement with an addition on the RHS into two
+    writes.
+
+    This operation is the opposite of the last two cases of `merge_writes`.
+
+    args:
+        stmt    - cursor pointing to the assign/reduce statement.
+
+    rewrite:
+        `a = b + c`
```

Comment:
We might also want to support
```
a = c
a += b
```
In simple cases, the `b` vs `c` computation ordering doesn't matter, but in more advanced cases (e.g. `c` is a value that's pre-computable), we might want to support this behavior. Plus it's a pretty simple extension to add.

Reply:
You can call commute_expr on the rhs and then use this operation to get that behavior.